### PR TITLE
Make band merch page configurable

### PR DESCRIPTION
**Pull request by binary1230**
Ability to turn on/off:
- Rock island text/dropdown option/validation
- Requirement that bands agree to sit at a table for 8 hours a day

NOTE: builds on an existing Pull Request (does not merge to master, merges to rename_rock_island_page branch)

### CODE REVIEW

**Comment by binary1230** [comment]
merge after https://github.com/magfest/production-config/pull/30

File: bands/configspec.ini [hunk 1]
```diff
@@ -27,12 +29,11 @@ bands = string(default="Band Management")
 [[band_merch]]
 no_merch = string(default="Not selling merch")
 own_table = string(default="Dedicated table")
-rock_island = string(default="Rock Island")
```
Comment (binary1230) [inline bands/configspec.ini]: this is the only questionable thing in this PR, having to remove this line.

My goal is to check if c.ROCK_ISLAND exists and then do different things in the template.

If configspec.ini has 'rock_island' in this enum, but the development.ini does NOT have it, c.ROCK_ISLAND will still created anyway.  There doesn't seem to be a way to remove/override part of an enum defined in configspec.ini. 

So the only way I could make this work was to remove rock_island here, and have prime and labs add it back in in their .ini files. (magstock does not add it back in)

File: bands/configspec.ini [hunk 2]
```diff
@@ -2,6 +2,8 @@
 band_email = string(default="MAGFest Music Department <music@magfest.org>")
 band_email_signature = string(default="- MAGFest Music Department")
 
+require_dedicated_band_table_presence = boolean(default=True)
```
Comment (kitsuta) [inline bands/configspec.ini]: Ehhh, I don't like this config option. This should be stuff that's done via model_checks, which is overridable in plugins. Take this config option out and I'll go ahead and do the refactoring necessary to make it work for Magstock.

File: bands/templates/bands/merch.html [hunk 3]
```diff
@@ -3,19 +3,30 @@
...
-we require that your table be staffed for at least <b>8 hours per day</b>.
+{% if c.ROCK_ISLAND %}
+<p>
+    It is recommended, but not required, for performers to use our Rock Island service in this area, as merchandise can
+    be sold more consistently, and for many more hours.
+</p>
+{% endif %}
 
-<br/> <br/>
+{% if c.REQUIRE_DEDICATED_BAND_TABLE_PRESENCE %}
```
Comment (kitsuta) [inline bands/templates/bands/merch.html]: Ah wait there's template stuff, nevermind I guess.

File: bands/configspec.ini (hunk 1 above)
Comment (kitsuta) [inline bands/configspec.ini]: Ah yes that's super true, this is the only way to do that. +1.

**Comment by kitsuta** [comment]
+1, ignore my comments about the new config option.

File: bands/configspec.ini (hunk 1 above)
Comment (EliAndrewC) [inline bands/configspec.ini]: As it happens, there's a poorly documented ("completely undocumented" would probably be more honest) feature in the way enums work, which is that if you set an option to the empty string, then our config parsing code will remove it from the options list.

In other words, we could keep this option here, and then if we set ``rock_island = ""`` in our puppet-managed ``development.ini``, then ``c.ROCK_ISLAND`` will still exist, but the value will be absent from ``c.BAND_MERCH_OPTS``.

So this change is fine, but not actually necessary.

I should probably document some more of this stuff :/

**Comment by EliAndrewC** [comment]
+1 (belatedly) with a note that part of this change isn't actually necessary.

**Comment by binary1230** [comment]
Ah it's not in the OPTS then, app in theory could test for that and set
some other config value or pass result into the template, good to know

**Pull request merged**